includes/admin/helper/class-wc-helper-product-install.php

```diff
@@ -303,44 +319,104 @@ private static function move_product( $product_id, $upgrader ) {
 		}
 
 		// TODO: handle theme.
-
-		return $upgrader->install_package( array(
-			'source'        => $steps[ $product_id ]['unpacked_path'],
-			'destination'   => WP_PLUGIN_DIR,
-			'clear_working' => true,
-			'hook_extra'    => array(
-				'type'   => 'plugin',
-				'action' => 'install',
+		return $upgrader->install_package(
+			array(
+				'source'        => $steps[ $product_id ]['unpacked_path'],
+				'destination'   => WP_PLUGIN_DIR,
+				'clear_working' => true,
+				'hook_extra'    => array(
+					'type'   => 'plugin',
+					'action' => 'install',
+				),
 			)
-		) );
+		);
 	}
 
 	/**
 	 * Activate product given its product ID.
 	 *
 	 * @param int $product_id Product ID.
+	 *
+	 * @return \WP_Error|null
 	 */
 	private static function activate_product( $product_id ) {
 		// Clear plugins cache used in `WC_Helper::get_local_woo_plugins`.
 		wp_clean_plugins_cache();
+		$filename = false;
 
-		$plugins = wp_list_filter(
-			WC_Helper::get_local_woo_plugins(),
-			array(
-				'_product_id' => $product_id,
-			)
-		);
+		// If product is WP.org one, find out its filename.
+		$folder_name = self::get_wporg_product_folder_name( $product_id );
+		if ( false !== $folder_name ) {
+			$filename = self::get_wporg_plugin_relative_path( $folder_name );
+		}
 
-		$filename = is_array( $plugins ) && ! empty( $plugins )
-			? key( $plugins )
-			: '';
+		if ( false === $filename ) {
+			$plugins = wp_list_filter(
+				WC_Helper::get_local_woo_plugins(),
+				array(
+					'_product_id' => $product_id,
+				)
+			);
+
+			$filename = is_array( $plugins ) && ! empty( $plugins )
+				? key( $plugins )
+				: '';
+		}
 
 		if ( empty( $filename ) ) {
 			return new WP_Error( 'unknown_filename', __( 'Unknown product filename.', 'woocommerce' ) );
 		}
 
-		// TODO: theme activation support
-
+		// TODO: theme activation support.
 		return activate_plugin( $filename );
 	}
+
+	/**
+	 * Get WP.org product filename.
+	 *
+	 * @param int $product_id Product ID.
+	 *
+	 * @return bool|string
+	 */
+	private static function get_wporg_product_folder_name( $product_id ) {
```

Comment:
prolly it's just me being pedantic. I think `dir` is more fit than `folder`. Not a biggie though.

Reply:
Method name was changed too. Thanks for your suggestion. The new name really suites better!